x/cdp/keeper/cdp.go

```diff
@@ -334,43 +352,52 @@ func (k Keeper) RemoveCdpOwnerIndex(ctx sdk.Context, cdp types.CDP) {
 		store.Delete(cdp.Owner)
 		return
 	}
-	store.Set(cdp.Owner, k.cdc.MustMarshalBinaryLengthPrefixed(updatedCdpIds))
+
+	updatedCdpIdsBytes, err := json.Marshal(updatedCdpIds)
+	if err != nil {
+		panic(err)
+	}
+	store.Set(cdp.Owner, updatedCdpIdsBytes)
 }
 
 // IndexCdpByCollateralRatio sets the cdp id in the store, indexed by the collateral type and collateral to debt ratio
 func (k Keeper) IndexCdpByCollateralRatio(ctx sdk.Context, collateralType string, id uint64, collateralRatio sdk.Dec) {
 	store := prefix.NewStore(ctx.KVStore(k.key), types.CollateralRatioIndexPrefix)
-	db, found := k.GetCollateralTypePrefix(ctx, collateralType)
+	_, found := k.GetCollateral(ctx, collateralType)
 	if !found {
 		panic(fmt.Sprintf("denom %s prefix not found", collateralType))
 	}
-	store.Set(types.CollateralRatioKey(db, id, collateralRatio), types.GetCdpIDBytes(id))
+	store.Set(types.CollateralRatioKey(collateralType, id, collateralRatio), types.GetCdpIDBytes(id))
 }
 
 // RemoveCdpCollateralRatioIndex deletes the cdp id from the store's index of cdps by collateral type and collateral to debt ratio
 func (k Keeper) RemoveCdpCollateralRatioIndex(ctx sdk.Context, collateralType string, id uint64, collateralRatio sdk.Dec) {
 	store := prefix.NewStore(ctx.KVStore(k.key), types.CollateralRatioIndexPrefix)
-	db, found := k.GetCollateralTypePrefix(ctx, collateralType)
+	_, found := k.GetCollateral(ctx, collateralType)
 	if !found {
 		panic(fmt.Sprintf("denom %s prefix not found", collateralType))
 	}
-	store.Delete(types.CollateralRatioKey(db, id, collateralRatio))
+	store.Delete(types.CollateralRatioKey(collateralType, id, collateralRatio))
 }
 
 // GetDebtDenom returns the denom of debt in the system
-func (k Keeper) GetDebtDenom(ctx sdk.Context) (denom string) {
+func (k Keeper) GetDebtDenom(ctx sdk.Context) string {
 	store := prefix.NewStore(ctx.KVStore(k.key), types.DebtDenomKey)
-	bz := store.Get([]byte{})
-	k.cdc.MustUnmarshalBinaryLengthPrefixed(bz, &denom)
-	return
+	bz := store.Get(types.DebtDenomKey)
+
+	var debtDenom gogotypes.StringValue
```

Comment:
nice, this is pretty interesting way to use the proto codec to handle primitives.